src/blueetl/cache.py

```diff
@@ -54,13 +56,33 @@ class CacheError(Exception):
     """Cache error raised when a read-only cache is written."""
 
 
-class LockManager:
+class LockManagerProtocol(Protocol):
+    """Lock Manager Interface."""
+
+    @property
+    @abstractmethod
+    def locked(self) -> bool:
+        """Return the lock status."""
+
+    @abstractmethod
+    def lock(self, mode: int) -> None:
+        """Lock."""
+
+    @abstractmethod
+    def unlock(self) -> None:
+        """Unlock."""
+
+
+class LockManager(LockManagerProtocol):
     """Lock Manager.
 
     On Linux, the flock call is handled locally, and the underlying filesystem (GPFS) does not get
     any notification that locks are being set. Therefore, GPFS cannot enforce locks across nodes.
     """
 
+    LOCK_EX = fcntl.LOCK_EX  # exclusive lock
```

Comment:
why are these class members?

Reply:
I created these aliases because I didn't want the parameters of the method `lock` to depend explicitly on fcntl constants, in case the implementation is going to change.
They could go also at the module level, but they are strictly related to this class. Alternatively, I could move all the LockManager stuff to a different module.

Reply:
> I didn't want the parameters of the method `lock` to depend explicitly on fcntl constants

I, ok. Makes sense.